LanguageFeatures/Augmentation-libraries/augmented_expression_A04_t04_lib.dart

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion The exact result of an `augmented` expression depends on what is
+/// being augmented, but it generally follows the same rules as any normal
+/// identifier:
+/// ...
+/// - Augmenting functions: When augmenting a function, augmented refers to the
+///   augmented function. Tear offs are not allowed, so this function must
+///   immediately be invoked.
+///
+/// @description Checks that it is a compile-time error to declare a local
+/// function named `augmented` inside of an augmenting function. Test a
```

Comment:
I guess this would be:
```suggestion
/// variable named `augmented` inside of an augmenting function. Test a
```
because pattern declarations introduce local variables.

Reply:
I haven't seen them all, yet, but I think this change is needed in all the pattern tests.